xfrd-tcp.c

```diff
@@ -639,6 +772,80 @@ tcp_conn_ready_for_reading(struct xfrd_tcp* tcp)
 	buffer_clear(tcp->packet);
 }
 
+#ifdef HAVE_TLS_1_3
+static int
+conn_write_ssl(struct xfrd_tcp* tcp, SSL* ssl)
+{
+	ssize_t sent;
+
+	if(tcp->total_bytes < sizeof(tcp->msglen)) {
+		uint16_t sendlen = htons(tcp->msglen);
+		// send
+		int request_length = sizeof(tcp->msglen) - tcp->total_bytes;
+		sent = SSL_write(ssl, (const char*)&sendlen + tcp->total_bytes,
+						 request_length);
+		switch(SSL_get_error(ssl,sent)) {
+			case SSL_ERROR_NONE:
+				if(request_length != sent)
+					log_msg(LOG_ERR, "xfrd: incomplete write of tls");
+				break;
+			default:
+				log_msg(LOG_ERR, "xfrd: generic write problem with tls");
```

Comment:
It would be nice to print more details, either from errno, or perhaps the function log_crypto_err() for ssl details, depending on what failed.  Perhaps also after other SSL_write and SSL_read and SSL_do_handshake failures.